swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -873,11 +878,15 @@ public Long getSelectedValue() {
 //					
 //				}
 //			}
-			if (settingSelectedItem && selectedItem != null) {
-				result = remodel.getMapping(selectedItem);
-			} else if (getSelectedIndex() != -1) {
-				result = remodel.getMapping(getSelectedIndex());
+			Object item = getSelectedItem();
+			if (item instanceof SSListItem) {
+				result = remodel.getMapping((SSListItem)item);
 			}
```

Comment:
I wonder about the evolution of settingSelectedItem/selectedItem. Now that mapping is part of SSListItem maybe that's why it's not needed. Or maybe because null are handled differently.

Reply:
settingSelectedItem was definitely a hack to deal with glazedlist. If the above works then maybe we can eliminate settingSelectedItem?